tests/test_cnes_sonar_scanner.py

```diff
@@ -0,0 +1,517 @@
+        # Hint: if this test fails, look for differences with: diff {tmp_file} {ref_file}
+        assert filecmp.cmp(os.path.join(cls._PROJECT_ROOT_DIR, tmp_file), os.path.join(cls._PROJECT_ROOT_DIR, ref_file))
+
+    @classmethod
+    def import_analysis_results(cls, project_name: str, project_key: str,
+        quality_profile: str, language_key: str, language_folder: str,
+        source_folder: str, rule_violated: str, expected_sensor: str,
+        expected_import: str, activate_rule: bool = False):
+        """
+        This function tests that the analysis results produced
+        by an analysis tool can be imported in SonarQube. The results
+        must be stored in the default files.
+
+        :param project_name: project name
+        :param project_key: project key
+        :param quality_profile: quality profile to use
+        :param language_key: language key
+        :param language_folder: folder to run the sonar-scanner in (relative to the root of the project)
+        :param source_folder: folder containing the source files (relative to the previous folder)
+        :param rule_violated: id of a rule violated by a source file
+        :param expected_sensor: line of output of the sonar-scanner that tells the import sensor is used
+        :param expected_import: line of output of the sonar-scanner that tells the result file was imported
+        :param activate_rule: True if if the rule violated needs to be activated in the Quality Profile for the import sensor to be run
+
+        Example (not a doctest):
+            rule_violated = "cppcheck:arrayIndexOutOfBounds"
+            expected_sensor = "INFO: Sensor C++ (Community) CppCheckSensor [cxx]"
+            expected_import = "INFO: CXX-CPPCHECK processed = 1"
+            self.import_analysis_results("CppCheck Dummy Project", "cppcheck-dummy-project",
+                "CNES_C_A", "c++", "tests/c_cpp", "cppcheck", rule_violated, expected_sensor, expected_import)
+        """
+        if activate_rule:
+            # Get the key of the Quality Profile to use
+            qp_key = requests.get(f"{cls.SONARQUBE_LOCAL_URL}/api/qualityprofiles/search?quality_profile={quality_profile}",
+                auth=("admin", cls.SONARQUBE_ADMIN_PASSWORD)).json()['profiles'][0]['key']
+            # Activate the rule in the Quality Profile to allow the Sensor to be used
+            requests.post(f"{cls.SONARQUBE_LOCAL_URL}/api/qualityprofiles/activate_rule",
+                auth=("admin", cls.SONARQUBE_ADMIN_PASSWORD),
+                data={
+                    "key": qp_key,
+                    "rule": rule_violated
+                })
+        # Create a project on SonarQube
+        errors = requests.post(f"{cls.SONARQUBE_LOCAL_URL}/api/projects/create",
+            auth=("admin", cls.SONARQUBE_ADMIN_PASSWORD),
+            data={
+                "name": project_name,
+                "project": project_key
+            }).json().get('errors', [])
+        assert not errors
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/cnescatlab/sonar-scanner/pullRequest?prid=6252273)